application-task-ui/src/main/resources/TaskManager/GanttMacro/UpdateService.xml

```diff
@@ -0,0 +1,260 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.4" reference="TaskManager.GanttMacro.UpdateService" locale="">
+  <web>TaskManager.GanttMacro</web>
+  <name>UpdateService</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>xwiki:TaskManager.GanttMacro</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity wiki="false"}}
+#macro (maybeSet $document $propName $propValue)
+  #if ($propValue)
+    $document.set($propName, $propValue)
+  #end
+#end
+
+##
+## This page acts as an endpoint for taskgantt macro to update the tasks.
+## It also contains some simple logic for updating the status based on progress.
+## This workaround is needed until XWIKI-20704 is fixed.
+##
+## Parameters:
+##   - taskDocument: document id (eg. wiki:Space.Page);
+##   - startDate, endDate: new times, given as unix epoch milliseconds;
+##   - progress: new progress value, 0-100 number;
+##
+## Returns a json with a 'success' property and, in the case of a failure, a 'reason' as defined in the translation keys.
+##
+
+
+#if ("$!request.outputSyntax" == 'plain')
+  #set ($discard = $response.setContentType('application/json'))
+#end
+
+#set ($taskDocument = $request.taskDocument)
+#if (!$xwiki.exists($taskDocument))
+  $jsontool.serialize({'success':false, 'reason':$services.localization.render('taskmanager.gantt.taskEdit.error.notExists')})
```

Comment:
Line too long

```
  $jsontool.serialize({
    'success':false,
    'reason':$services.localization.render('taskmanager.gantt.taskEdit.error.notExists')
  })
```

I see the same for the other ` $jsontool.serialize` calls below. It's also easier to read when there is a structure and not in just one line